repos/system_upgrade/common/actors/checkldsoconf/libraries/checkldsoconfiguration.py

```diff
@@ -0,0 +1,127 @@
+import glob
+import os
+
+from leapp import reporting
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import InstalledRedHatSignedRPM
+
+LD_SO_CONF_DIR = '/etc/ld.so.conf.d'
+LD_SO_CONF_MAIN = '/etc/ld.so.conf'
+LD_SO_CONF_DEFAULT_INCLUDE = 'ld.so.conf.d/*.conf'
+LD_SO_CONF_COMMENT_PREFIX = '#'
+FMT_LIST_SEPARATOR = '\n    - '
+
+
+def _read_file(file_path):
+    with open(file_path, 'r') as fd:
+        return fd.readlines()
+
+
+def _report_custom_ld_so_configuration(summary):
+    reporting.create_report([
+        reporting.Title(
+            'Third party libraries linked with ld.so.conf are not supported.'
+        ),
+        reporting.Summary(summary),
+        reporting.Remediation(hint=('Remove the custom ld.so configuration and apply the changes '
+                                    'using the ldconfig command.')),
+        reporting.RelatedResource('file', '/etc/ld.so.conf'),
+        reporting.RelatedResource('directory', '/etc/ld.so.conf.d'),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Groups([reporting.Groups.OS_FACTS]),
+    ])
+
+
+def _is_modified(config_path, package_name):
+    """ Decide if the configuration file was modified based on the package it belongs to. """
+    is_modified = False
+    result = run(['rpm', '-Va', package_name], checked=False)
```

Comment:
Is there a reason why you use `rpm -Va`? Although this does work I think a better approach would be to use `rpm -V <package name>` or, even better, `rpm -Vf <file name>` which would save you some of the parsing. See the package selection options in `man rpm`.

Reply:
Also when I try `rpm -Va` with the full kernel package name instead of `kernel-core` it doesn't detect a modified `/etc/ld.so.conf.d/kernel-4.18.0-477.10.1.el8_8.x86_64.conf` whereas both `rpm -V` and `rpm -Vf` do.

Reply:
Ou yes, `-Vf` seems like the way to go, I just can't read and somehow mismatched those...

Reply:
Fixed thanks!